### PR TITLE
Pull request allows to choose branch

### PR DESCRIPTION
**Pull request by linquize**
Pull request allows to choose branch

### CODE REVIEW

File: routers/repo/compare.go [hunk 1]
```diff
@@ -135,12 +135,12 @@ func ForkDiff(ctx *middleware.Context, beforeCommitId, afterCommitId string) {
 		return
 	}
 
-	afterCommit, err := repo.GetCommitIdOfRef("refs/remotes/upstream/" + afterBranch)
+	afterCommit, err := repo.GetCommitOfBranch(afterBranch)
```
Comment (lunny) [inline routers/repo/compare.go]: This line is different meaning from the before line?

Comment (linquize) [inline routers/repo/compare.go]: This is because the contributor can create a new branch, which does not exist in upstream repository.

**Comment by lunny** [comment]
@linquize, would you like to be as a member of gitea team?

File: public/ng/js/gitea.js [hunk 2]
```diff
@@ -1019,6 +1019,18 @@ function initTimeSwitch() {
 }
 
 function initDiff() {
+    $('#pr-upstream-repo').on("change", function () {
```
Comment (tboerger) [inline public/ng/js/gitea.js]: Just write this once and write the sepectors coa seperated.

File: modules/markdown/markdown.go [hunk 3]
```diff
@@ -94,6 +95,64 @@ var (
 	// FIXME: this pattern matches pure numbers as well, right now we do a hack to check in RenderSha1CurrentPattern
 	// by converting string to a number.
 	Sha1CurrentPattern = regexp.MustCompile(`\b[0-9a-f]{40}\b`)
+
+	WikiLinkPattern = regexp.MustCompile(`(\[\[.*\]\]\w*)`)
+
+	/*
```
[the remark this answers is not in the document]
Comment (bkcsoft) [reply modules/markdown/markdown.go]: move those patterns into a unit-test instead ;)

File: modules/markdown/markdown.go [hunk 4]
```diff
@@ -130,46 +193,50 @@ func (r *Renderer) AutoLink(out *bytes.Buffer, link []byte, kind int) {
...
+	if bytes.HasPrefix(text, []byte("<p>")) {
+		prefix = "<p>"
+	}
 	switch {
-	case bytes.HasPrefix(text, []byte("[ ] ")):
-		text = append([]byte(`<input type="checkbox" disabled="" />`), text[3:]...)
-	case bytes.HasPrefix(text, []byte("[x] ")):
-		text = append([]byte(`<input type="checkbox" disabled="" checked="" />`), text[3:]...)
+	case bytes.HasPrefix(text, []byte(prefix+"[ ] ")):
+		text = append([]byte(`<div class="ui fitted read-only checkbox"><input type="checkbox" /><label /></div>`), text[3+len(prefix):]...)
```
[the remark this answers is not in the document]
Comment (andreynering) [reply modules/markdown/markdown.go]: > you mean markdown editor frontend?

Yes

File: public/js/index.js [hunk 5]
```diff
@@ -594,6 +594,7 @@ function initWikiForm() {
                         function (data) {
                             preview.innerHTML = '<div class="markdown">' + data + '</div>';
                             emojify.run($('.editor-preview')[0]);
+                            $('.editor-preview').autolink();
```
[the remark this answers is not in the document]
Comment (tboerger) [reply public/js/index.js:616]: IMHO that should be done on the backend, otherwise it will never get indexed for public wikis

**Comment by lunny** [comment]
OK. LGTM

**Pull request merged**